NuRadioMC/examples/simulate_effective_ice_temperature/ice_integrator.py

```diff
@@ -0,0 +1,258 @@
+    model : str
+        Ice attenuation model to use
+
+    Returns
+    -------
+    eff_temperature : float
+        Effective temperature at the antenna
+    """
+    # get the track for respective zenith direction
+    # (distance vs depth)
+    _, radius, depth_raytracing, distance_raytracing, reflected_power, _ = get_trace(zenith, z_antenna)
+
+    # get the temperature at each of the track points
+    depth_interp = interp1d(distance_raytracing, depth_raytracing, fill_value="extrapolate")
+    power_interp = interp1d(distance_raytracing, reflected_power, fill_value="extrapolate")
+
+    distance = np.linspace(0, 30000, 300000) * units.m
+    d_distance = distance[1] - distance[0]
+
+    depth = depth_interp(distance)
+    ref = power_interp(distance)
+    # manually set cap stuff
+    ref[depth < z_min] = 0
+    depth[depth < z_min] = z_min
+
+    l_att = attenuation.get_attenuation_length(depth, frequency=freq, model=model)
+    meaned_l_att = np.cumsum(l_att) / np.cumsum(np.ones_like(distance))
+
+    att_factor = np.exp(-distance / meaned_l_att)
+
+    assert model.startswith("GL"), "Only the Greenland ice model is supported for now (because the GRIP temperature model is hardcoded.)"
+    temp_env = attenuation.get_grip_temperature(depth)  # already in kelvin
+
+    return np.sum((ref * att_factor * temp_env / l_att * d_distance))
+
+
+def get_eff_temperature(z_antenna=-100, n_theta=100, plot=False, fname=None):
+    import time
+    t0 = time.time()
+    thetas = np.linspace(0, np.pi, n_theta)
+
+    eff_temperatures = []
+    for theta in thetas:
+        eff_temperatures.append(float(temperature_integral(theta, z_antenna)))
+
+    data = {
+        "z_antenna": z_antenna,
+        "theta": thetas.tolist(),
+        "eff_temperature": eff_temperatures
+    }
+
+    if fname is None:
+        fname = f"eff_temperature_{z_antenna}_ntheta{n_theta}.json"
+
+    with open(fname, "w") as f:
+        json.dump(data, f, indent=4)
+    print(time.time() - t0)
+
+    if plot:
+        fig, ax = plt.subplots()
+        ax.plot(np.rad2deg(thetas), eff_temperatures)
+        ax.set_xlabel("theta / deg")
+        ax.set_ylabel("temperature / K")
+        plt.show()
+
+def plot_ray_paths(z_antenna=-100, n_theta=100):
+    thetas = np.linspace(0, np.pi, n_theta)
+
+    import matplotlib as mpl
+    cmap = plt.get_cmap('plasma')
+
+    norm = mpl.colors.Normalize(vmin=0, vmax=180)
+    sm = mpl.cm.ScalarMappable(norm=norm, cmap=cmap)
+
+    fig, ax = plt.subplots()
+    for theta in thetas:
+        _, x, z, _, _, _ = get_trace(theta, z_antenna)
+        ax.plot(x, z, color=sm.to_rgba(np.rad2deg(theta)), alpha=0.8)
+    cb = plt.colorbar(sm, ax=ax, pad=0.02)
+    cb.set_label("theta / deg")
+
+    ax.set_xlabel("x / m")
+    ax.set_ylabel("z / m")
+    ax.set_xlim(-10, 1000)
+    fig.tight_layout()
+    plt.savefig("plot.pdf")
+
+
+if __name__ == "__main__":
+
+    get_eff_temperature(fname="eff_temperature_grip_GL3_400MHz.json")
```

Comment:
you might want to consider setting the keyword argument `plot=True` for the main

Reply:
done

Reply:
thanks